app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/steps/select_attributes.js

```diff
@@ -114,6 +121,54 @@ define([
             this.selectedAttributes(labels.join(', '));
         },
 
+        /**
+         * @param {*} selected
+         */
+        doSelectedAttributesCodes: function (selected) {
+            var code = [];
+            this.selected = selected;
+            _.each(selected, function (attributeId) {
+                var attribute;
+
+                if (!this.attributesCode[attributeId]) {
+                    attribute = _.findWhere(this.multiselect().rows(), {
+                        'attribute_id': attributeId
+                    });
+                    if (attribute) {
+                        var chosen = {
+                            'id': attribute.attribute_id,
+                            'attribute_code': attribute.attribute_code,
+                            'attribute_label': attribute.frontend_label,
+                            'label': '',
+                            'value': '0'
+                        }
+                        var chose = [];
+                        chose.push(chosen);
+                        var newatrr = {
+                            'id': attribute.attribute_id,
+                            'code': attribute.attribute_code,
+                            'label': attribute.frontend_label,
+                            'chosen': chose,
+                            'position': 0
+                        }
+                        this.attributesCode[attribute['attribute_id']] = newatrr;
+                    }
+                }
+                code.push(this.attributesCode[attributeId]);
+
+            }.bind(this));
+            this.attributes(code);
+        },
+
+        doShowAddProductButton: function (selected) {
+            if (selected.length > 0) {
+                this.disabledButton(false);
+            } else {
+                this.disabledButton(true);
+            }
```

Comment:
I think we can simplify this statement by using the following instead of if-else statement.
```
            this.disabledButton(!selected.length);
```

Reply:
fixed and committed @dmytro-ch